src/client/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -496,31 +498,46 @@ export class KernelDebugAdapter implements DebugAdapter, IKernelDebugAdapter, ID
         await this.dumpCell(cell.document.uri.toString());
 
         if (this.configuration.__mode === KernelDebugMode.RunByLine) {
-            const initialBreakpoint: DebugProtocol.SourceBreakpoint = {
-                line: 1
-            };
-            const splitPath = cell.notebook.uri.path.split('/');
-            const name = splitPath[splitPath.length - 1];
-            const message: DebugProtocol.SetBreakpointsRequest = {
-                seq: seq + 1,
-                type: 'request',
-                command: 'setBreakpoints',
-                arguments: {
-                    source: {
-                        name: name,
-                        path: cell.document.uri.toString()
-                    },
-                    lines: [1],
-                    breakpoints: [initialBreakpoint],
-                    sourceModified: false
+            const textLines = cell.document.getText().splitLines({ trim: false, removeEmptyEntries: false });
+            const lineList: number[] = [];
+            parseForComments(
+                textLines,
+                () => noop(),
+                (s, i) => {
+                    if (s.trim().length !== 0) {
+                        lineList.push(i);
+                    }
                 }
-            };
-            this.sendRequestToJupyterSession(message);
+            );
+            lineList.sort();
+
+            if (lineList.length !== 0) {
+                const initialBreakpoint: DebugProtocol.SourceBreakpoint = {
+                    line: lineList[0] + 1
+                };
+                const splitPath = cell.notebook.uri.path.split('/');
+                const name = splitPath[splitPath.length - 1];
```

Comment:
Is this the name of the notebook? If it is, then its clearer to use fs.baseName or similar API on `cell.notebook.uri.fspath`, instead of splitting with `/`.
E.g. on windows it could be `\`.